### DISCUSSION TITLE
How to syntax-highlight a fish command?

### DISCUSSION THREAD

**Discussion by JanPokorny** [fish-shell/fish-shell#9793] [Q&A] [answered]
I use `fzf` for `CTRL+R`. It's really nice. But it does not support syntax highlighting, and I like syntax highlighting, so I decided to hack it together.

How does `fish` do the syntax highlighting, though? Is there a builtin command that will take a string and highlight it? I know I could use `bat -l fish`, but it uses a different color palette and is an external program. I would prefer to just let `fish` highlight the command, if there is a way to programatically call the highlighter.

**Comment by faho** [comment, marked as answer]
`fish_indent --ansi` will highlight any command you give it via stdin. It is a real binary on your system. It will indent all the input as well, use `--no-indent` to suppress that (it will still reformat to one job per line tho).

Or you could update to fish 3.6 and try the new builtin history pager, which highlights automatically.

**Reply by JanPokorny** [reply]
Splendid, exactly what I asked for, thank you very much!

I know about the new pager, but I prefer fuzzy matching. I'll see what I can do with this, thanks again!

**Reply by faho** [reply]
The pager does fuzzy matching - as in "find these characters in that order, but possibly with things in-between" - `ecfrd` will find `echo fus roh dah`

**Reply by JanPokorny** [reply]
@faho That does not seem to work for me:

![image](https://github.com/fish-shell/fish-shell/assets/4580066/50372c7a-e751-4171-9e4d-26519041124e)

It only does an exact match:
![image](https://github.com/fish-shell/fish-shell/assets/4580066/0d700126-6f99-4a5a-bd33-ba06613f12ff)

I seem to be using the latest Fish (from Homebrew): 
![image](https://github.com/fish-shell/fish-shell/assets/4580066/99115942-1b19-49a6-a230-4be494320ebd)

**Reply by faho** [reply]
Aaah, yeah that's #9476, coming in the next release.

**Reply by JanPokorny** [reply]
BTW, is it possible to invoke this new pager programatically? (Like with fzf, for example with a custom list of folders from zoxide?)

**Reply by faho** [reply]
No. That's #2805.
